src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterDataTest.java

```diff
@@ -82,6 +83,32 @@ public SimpleFeature apply(ImportTask task, DataStore dataStore, SimpleFeature o
         }
     }
     
+    @Test
+    public void testUploadRootExternalProps() throws Exception {
+        // On a brand new data folder, the directory may not exists until the Importer has been invoked the first time
+        File dirFromProperties = Resources.directory(Resources.fromPath("props_uploads"));
+        
+        // Read the upload root folder and creates it if does not exists
+        importer.getUploadRoot();
+        
+        // Read the folder again...
+        dirFromProperties = Resources.directory(Resources.fromPath("props_uploads"));
+        // ... and ensure it is the same as defined on the .properties file
+        assertEquals(dirFromProperties, importer.getUploadRoot());
+        
+        // Let's now override the external folder through the Environment variable. This takes precedence on .properties
+        System.setProperty(Importer.UPLOAD_ROOT_KEY, "env_uploads");
```

Comment:
A test setting an env variable should be clearing it at the end (finally block here, or @AfterTest method)

Reply:
Right, I will fix it.